rts/Sim/Units/UnitDef.cpp

```diff
@@ -694,6 +693,33 @@ UnitDef::UnitDef(const LuaTable& udTable, const std::string& unitName, int id)
 			}
 		}
 	}
+	{
+		//canAssist should default to true for mobile builders but should
+		//default to false for factories.
+		//
+		//To determine if this unit is a factory, previous unitDef 
+		//features need to be parsed, which is why canAssist is defined
+		//last.
+		//
+		//Note that a mobile builder with canAssist = false will be able
+		//to place a nanoframe and pour buildpower into it, but will not
+		//be able to resume building if interrupted for any reason (with
+		//the exception of the wait command). It will be unable to pour
+		//buildpower into a nanoframe placed by another unit. It will be
+		//unable to repair an incomplete nanoframe or place a nanoframe
+		//on top of an existing nanoframe, even if it is the exact same
+		//structure in the exact same location.
+		
+		//Using the following definitions from UnitDef.h:
+		//isImmobileUnit() = (pathType == -1U && !canfly && speed <= 0.0f); 
+		//isBuildingUnit() = (IsImmobileUnit() && !yardmap.empty());;
+		//isBuilderUnit() = (builder && buildSpeed > 0.0f && buildDistance > 0.0f);;
+		//IsFactoryUnit() = (IsBuilderUnit() &&  IsBuildingUnit());
+		canAssist = udTable.GetBool("canAssist", (builder && //!IsFactoryUnit()
```

Comment:
I think ideally it would stay assigned as-is (in the block above) and this here would just do
```
if (IsFactoryUnit())
   canAssist = false;
```
If `IsFactoryUnit` can't be called at this point then just move it down to some place it becomes possible.

Reply:
I don't usually program much in C++, but I don't remember being able to call an object's member functions like [IsFactoryUnit()](https://github.com/beyond-all-reason/spring/blob/6cd4504661bd8e0753ada4231842c92463e120b4/rts/Sim/Units/UnitDef.h#L66) until after it is initialized, right?

Reply:
You can call a member function just fine, it's even already done in this constructor:
https://github.com/beyond-all-reason/spring/blob/6cd4504661bd8e0753ada4231842c92463e120b4/rts/Sim/Units/UnitDef.cpp#L335

But perhaps it would be good to put this logic in the sanitisation section added in #1277 that happens after everything is loaded.

Reply:
Nice, thanks for all the teaching and help.

I like the idea of including it in the sanitization section of UnitDefHandler. I seems likely that other sanitization-like code will be needed in the future, so in the latest commit I renamed UnitDefHandler.SanitizeCorpseNames() to UnitDefHandler.SanitizeUnitDefs(), restored the original code for canAssist in UnitDef and added documentation comments, and then within SanitizeUnitDefs() I set canAssist to false for factories.

Please check the latest commit and let me know if that's closer to what you want.

Reply:
Looks great, thanks.